docs/src/main/asciidoc/spanner.adoc

```diff
@@ -87,44 +218,50 @@ public class MyApplication {
+	    studentRepository.findTop3DistinctByIdAndNameOrAgeOrderByAgeDesc("someId", "bob", 12L);
 
 	  List<SimpleStudent> ssList2 = studentRepository.fetchBySomeQuery("some argument");
 	}
 }
 
 ----
 
-The https://docs.spring.io/spring-data/commons/docs/current/reference/html/#repositories.query-methods[query methods]
-in the `StudentRepository` are generated based on the convention of their names
-except for `fetchBySomeQuery`, which is defined in the `spanner-named-queries.properties` file
-like `SimpleStudent.fetchBySomeQuery=SELECT * FROM students WHERE name = @someArg`.
+=== Query methods
 
-When defining `StudentRepository`, it is important to note that it extends the
-`PagingAndSortingRepository`, which is supported by Spring Cloud GCP Data Spanner. The first type
-argument refers to the underlying entity, and the second type parameter refers to the `@Id` type
-of that entity. Id types must correspond to one of the https://cloud.google.com/spanner/docs/data-types[supported primary key types in Spanner].
 
-=== Configuration
+==== Resolving methods by name
 
-The Spring Boot Starter for Google Cloud Data Spanner provides the following configuration options:
+In the example above, the {spring-data-commons-ref}/#repositories.query-methods[query methods]
+in `TradeRepository` are generated based on the convention of their names.
 
-|===
-| Name | Description | Optional | Default value
-| `spring.cloud.gcp.spanner.projectId` | the GCP project ID to use | Yes | provided by Spring Cloud GCP
-| `spring.cloud.gcp.spanner.instanceId` | Spanner instance to use | No |
-| `spring.cloud.gcp.spanner.database` |
-Spanner database to use | No |
-|===
+`List<Trade> findByAction(String action)` would translate to a `SELECT * FROM trades WHERE action = ?`.
+
+You can think about the little bit extreme `List<Trade> findTop3DistinctByActionAndSymbolOrTraderIdOrderBySymbolDesc(String action, String symbol, String traderId);` as
+
+[source, sql]
+----
+SELECT DISTINCT * FROM trades
+WHERE ACTION = ? AND SYMBOL = ? AND Or TRADER_ID = ?
+ORDER BY SYMBOL DESC
+LIMIT 3
+----
+
+==== Mapping SQL to repository methods using properties file
+
+
+In the example above for `List<Trade> fetchByActionNamedQuery(String action)` does not match the naming convention.
+The query for it is defined in the `spanner-named-queries.properties` file
+
+[source, properties]
+----
+Trade.fetchByActionNamedQuery=SELECT * FROM trades WHERE trades.action = @tag0`
+----
 
-Custom queries can be defined in properties files provided to the annotation
-`@EnableSpannerRepositories(namedQueriesLocation = "classpath:/[my-named-queries.properties file]")`.
-These query settings take the form `[ClassName].[QueryMethodName]=[query string with tags]`, where Spanner
-tags for parameters have the format `@[tagName]`. The actual names of the tags do not matter as long
-their order matches that defined in the repository interface.
+==== Mapping SQL to repository methods using annotation
```

Comment:
Again, if I understand correctly, this has nothing to do with our project, so we shouldn't be in the business of explaining it.

Reply:
I think this section is referring to the @Query annotation that is coming soon, which we will need to implement.

Reply:
Yes, I did mean the `@Query` annotation. It is coming soon, but I'll front it in the doc for better understandability.